### PR TITLE
Multi openai endpoint support

### PR DESCRIPTION
**Pull request by NeonBohdan**
# Description
Supports:
- Specifying list of endpoints and list of api keys
- Instead of specifying system prompt now pesonas referenced by their name described in `config.yaml`
- User can specify `model name, persona name` inside persona description

# Issues
Limitations:
- Service uses old generation parameters, to simplify reproducibility debugging
- Service uses two list for ulrs and keys instead of single dict
- Instead of error service fallbacks to first model and first persona if specified incorrectly
- revision isn't optional here, but it's ok, since brainforge deployment obligates specifying revision
# Other Notes
- [x] MQ communication is working
- [x] Responses in Klat `!prompt` are the same

### CODE REVIEW

File: neon_llm_vllm/vllm.py [hunk 1]
```diff
@@ -76,54 +106,72 @@ def context_depth(self):
...
         vllm_model_name = models.data[0].id
 
         model_name, *suffix = vllm_model_name.split("@")
         revision = dict(enumerate(suffix)).get(0, None)
 
-        return ModelMetadata(
+        tokenizer = self.tokenizer(model_name, revision)
+        personas = self.get_personas(model_name, revision)
+
+        return vllm_model_name, ModelMetadata(
```
Comment (NeonKirill) [inline neon_llm_vllm/vllm.py]: please update to return only ModelMetadata

**Comment by NeonBohdan** [comment]
Validated:
- description parsing to model and persona 
- model and persona switch per request
- system prompt extraction
- ask and vote same results

**Comment by NeonBohdan** [comment]
Multi model vllm deployed to alpha

Vote works as intended but produces controversial results 
Need feedback from more people (if it need to be fixed)

File: neon_llm_vllm/vllm.py [hunk 2]
```diff
@@ -41,11 +44,38 @@
...
 }
 
+class PileConfig(BaseModel):
+    persona2system: Dict[str, str]
+
+# TODO: use this params instead of COMPLETION_GENERATION_EXTRA_ARGUMENTS
+class InferenceConfig(BaseModel):
+    streaming: bool = True
+    temperature: float = 0.0
+    repetition_penalty: float = 1.05
```
Comment (NeonDaniel) [inline neon_llm_vllm/vllm.py:54]: Related to https://github.com/NeonGeckoCom/NeonLLM-Forge/issues/84

File: neon_llm_vllm/vllm.py [hunk 3]
```diff
@@ -76,54 +106,72 @@ def context_depth(self):
...
 
     @property
     def llm_model_name(self) -> str:
-        return self.model_metadata.vllm_model_name
+        pass
 
     @property
     def _system_prompt(self) -> str:
-        return self.role
+        pass
```
Comment (NeonDaniel) [inline neon_llm_vllm/vllm.py:132]: Should `system_prompt` be removed/refactored in the base class? In general, I don't know that we want to specify a new instance of NeonLLM for each persona.

**Pull request merged**